tests/lib/rules/require-description.js

```diff
@@ -31,6 +31,8 @@ tester.run("require-description", rule, {
         "/* eslint-disable-next-line -- description */",
         "// eslint-disable-line eqeqeq -- description",
         "// eslint-disable-next-line eqeqeq -- description",
+        `// some comment above
+// eslint-disable-next-line eqeqeq`,
```

Comment:
Maybe add something like

```
/* 
  * Some multiline comment
  */
// eslint-disable-next-line eqeqeq
```

... as well, just to make sure it at least is covered by tests. I can see that you look at `comment.loc.end.line` so it should work.